# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: layout/build/nsLayoutStatics.cpp
# Commit: dd9794f7d400
# Full Hash: dd9794f7d400231a0a06a41ff56e671f05fb825e
# Author: Masayuki Nakano <masayuki@d-toybox.com>
# Date: 2019-06-10 21:47:35
# Regressor Bug: 1529884
# File Overlap Count: 1
# Description:
#   Bug 1529884 - part 1: Make Document use hashtable to map HTML command and internal command data r=smaug
#   
#   Currently, `Document` converts HTML command (e.g., used by `execCommand()`) to
#   internal XUL command with array in the global space.  However, it requires scan
#   of the array for every command access.
# ==============================================================================

diff -r 66cd074a9182 -r dd9794f7d400 layout/build/nsLayoutStatics.cpp
--- a/layout/build/nsLayoutStatics.cpp	Mon Jun 10 10:23:26 2019 +0000
+++ b/layout/build/nsLayoutStatics.cpp	Mon Jun 10 10:23:59 2019 +0000
@@ -113,6 +113,7 @@
 #include "mozilla/HTMLEditorController.h"
 #include "mozilla/ServoBindings.h"
 #include "mozilla/StaticPresData.h"
+#include "mozilla/dom/Document.h"
 #include "mozilla/dom/IPCBlobInputStreamStorage.h"
 #include "mozilla/dom/WebIDLGlobalNameHash.h"
 #include "mozilla/dom/U2FTokenManager.h"
@@ -336,6 +337,7 @@
     URLExtraData::ReleaseDummy();
   }
 
+  Document::Shutdown();
   nsMessageManagerScriptExecutor::Shutdown();
   nsFocusManager::Shutdown();
 #ifdef MOZ_XUL
